File: app/src/main/java/com/amperas17/mangaedenapp/ui/gallery/viewmodel/ChapterPagesViewModel.java
package com.amperas17.mangaedenapp.ui.gallery.viewmodel;

import android.arch.lifecycle.LiveData;

import com.amperas17.mangaedenapp.data.ChapterRepository;
import com.amperas17.mangaedenapp.model.Resource;
import com.amperas17.mangaedenapp.model.page.Page;
import com.amperas17.mangaedenapp.ui.BaseViewModel;

import java.util.ArrayList;


public class ChapterPagesViewModel extends BaseViewModel<ArrayList<Page>> {

    private ChapterRepository repository;

    public LiveData<Resource<ArrayList<Page>>> getResource() {
        if (repository == null) {
            repository = new ChapterRepository(this);
        }
        return super.getResource();
    }

    private void loadData(String chapterID) {
        repository.callData(chapterID);
    }

    public void startLoading(String mangaID) {
        super.startLoading();
        loadData(mangaID);
    }

    public void stopLoading() {
        super.stopLoading();
        repository.cancelDataRequest();
    }
}

File: app/src/main/java/com/amperas17/mangaedenapp/ui/gallery/viewmodel/ChapterPagesViewModel.kt
package com.amperas17.mangaedenapp.ui.gallery.viewmodel

import com.amperas17.mangaedenapp.data.ChapterRepository
import com.amperas17.mangaedenapp.model.page.Page
import com.amperas17.mangaedenapp.ui.BaseViewModel

import java.util.ArrayList


class ChapterPagesViewModel : BaseViewModel<ArrayList<Page>>() {

    private val repository = ChapterRepository(this)

    private fun loadData(chapterID: String) {
        repository.callData(chapterID)
    }

    fun startLoading(mangaID: String) {
        super.startLoading()
        loadData(mangaID)
    }

    override fun stopLoading() {
        super.stopLoading()
        repository.cancelDataRequest()
    }
}
